packages/x-date-pickers/src/MonthCalendar/tests/MonthCalendar.test.tsx

```diff
@@ -26,14 +26,27 @@ describe('<MonthCalendar />', () => {
     expect(onChange.args[0][0]).toEqualDateTime(new Date(2019, 1, 2));
   });
 
-  it('should select start of month without time when no initial value is present', () => {
-    const onChange = spy();
-    render(<MonthCalendar onChange={onChange} />);
+  describe('with fake timers', () => {
+    // TODO: temporary for vitest. Can move to `vi.useFakeTimers`
+    let timer: SinonFakeTimers | null = null;
 
-    fireEvent.click(screen.getByRole('radio', { name: 'February' }));
+    beforeEach(() => {
+      timer = useFakeTimers({ now: new Date(2019, 0, 1), toFake: ['Date'] });
+    });
 
-    expect(onChange.callCount).to.equal(1);
-    expect(onChange.args[0][0]).toEqualDateTime(new Date(2019, 1, 1, 0, 0, 0));
+    afterEach(() => {
+      timer?.restore();
+    });
+
+    it('should select start of month without time when no initial value is present', () => {
+      const onChange = spy();
+      render(<MonthCalendar onChange={onChange} />);
+
+      fireEvent.click(screen.getByRole('radio', { name: 'February' }));
```

Comment:
Nitpick: was there a roadblock for using `userEvent` here? 🤔

Reply:
Didn't try. I mostly try to change stuff where it is causing issues.